src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -90,12 +91,12 @@ public Car update(Car car) {
             updateStatement.setLong(2, car.getManufacturer().getId());
             updateStatement.setLong(3, car.getId());
             updateStatement.executeUpdate();
-            deleteCarDrivers(car.getId());
-            saveDrivers(car);
         } catch (SQLException throwable) {
             throw new DataProcessingException("Can't update "
                     + car + " in DB.", throwable);
         }
+        deleteCarDrivers(car.getId());
+        saveDrivers(car);
```

Comment:
SAme here add helper boolean or int variable and assign it value of updateStatement.executeUpdate();
then check if the was deleted the delete car and saveDrivers